.github/workflows/ci_nix.yml

```diff
@@ -0,0 +1,18 @@
+name: "Nix checks"
+on:
+  pull_request:
+jobs:
+  minimal:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: nixbuild/nix-quick-install-action@v21
+        with:
+          nix_conf: experimental-features = nix-command flakes
+      - name: nix flake check
+        run: nix flake check
+      - name: nix fmt
+        run: |
+          nix fmt $(find . -maxdepth 2 -name \*nix -type f)
```

Comment:
We probably don't want `-maxdepth` here. I have a draft PR that adds nix files in `codebuild/bin`. They should probably go somewhere else, but no reason to limit the formatter.

Reply:
It was slow on my host; removed in the latest commit.  I think the convention is to have a top level `nix` folder, but another PR.